src/services/studio/studio.service.ts

```diff
@@ -125,33 +124,49 @@ class StudioService {
         });
     }
 
-    private deleteFileIfTypeScenario(packageZip: AdmZip, entry: IZipEntry): void {
-        if (entry.entryName.startsWith("nodes/") && entry.entryName.endsWith(".yml")) {
-            const node: NodeExportTransport = parse(entry.getData().toString());
-            if (node.type === "SCENARIO") {
-                packageZip.deleteFile(entry);
+    private deleteScenarioAssets(packageZip: AdmZip): void {
+        packageZip.getEntries().forEach(entry => {
+            if (entry.entryName.startsWith("nodes/") && entry.entryName.endsWith(".yml")) {
+                const node: NodeExportTransport = parse(entry.getData().toString());
+                if (node.type === "SCENARIO") {
+                    packageZip.deleteFile(entry);
+                }
             }
-        }
+        })
     }
 
-    private fixConnectionVariablesIfRootNodeFile(packageZip: AdmZip, entry: IZipEntry, zipName: string, exportedVariables: VariableManifestTransport[]): void {
-        if (entry.name === "package.yml") {
-            const packageKeyAndVersion = zipName.replace(".zip", "").split("_");
-            const connectionVariablesByKey = this.getConnectionVariablesByKeyForPackage(packageKeyAndVersion[0], packageKeyAndVersion[1], exportedVariables);
+    private fixConnectionVariablesForRootNodeFiles(packageZip: AdmZip, zipName: string, exportedVariables: VariableManifestTransport[]): void {
+        packageZip.getEntries().forEach(entry => {
+            if (entry.name === "package.yml") {
```

Comment:
I think we could use `find` here to find the `package.yml` file, and then perform the rest of the changes.